src/vi/mimihentai/src/eu/kanade/tachiyomi/extension/vi/mimihentai/MiMiHentai.kt

```diff
@@ -0,0 +1,201 @@
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS", Locale.US)
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(
+        apiUrl.toHttpUrl().newBuilder().apply {
+            addPathSegments("tatcatruyen")
+            addQueryParameter("page", page.toString())
+            addQueryParameter("sort", "updated_at")
+        }.build(),
+        headers,
+    )
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val res = response.parseAs<MangaDTO>()
+        val manga = res.data.map {
+            SManga.create().apply {
+                title = it.title
+                thumbnail_url = it.coverUrl
+                setUrlWithoutDomain("$baseUrl/g/${it.id}")
+            }
+        }
+        val hasNextPage = res.currentPage != res.totalPage
+        return MangasPage(manga, hasNextPage)
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("/")
+        val url = apiUrl.toHttpUrl().newBuilder()
+            .addPathSegment("gallery")
+            .addPathSegment(id)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val res = response.parseAs<List<ChapterDTO>>()
+        return res.map { chapterDTO ->
+            SChapter.create().apply {
+                name = chapterDTO.title
+                date_upload = dateFormat.tryParse(chapterDTO.createdAt)
+                setUrlWithoutDomain("/api/v1/manga/chapter?id=" + chapterDTO.id)
+            }
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("/")
+        val url = apiUrl.toHttpUrl().newBuilder()
+            .addPathSegment("info")
+            .addPathSegment(id)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val res = response.parseAs<Manga>()
+        return res.let { manga ->
+            SManga.create().apply {
+                description = manga.description
+                author = manga.authors.joinToString { i -> i.name }
+                genre = manga.genres.joinToString { i -> i.name }
+                thumbnail_url = manga.coverUrl
+                title = manga.title
+            }
+        }
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val json = JSONObject(response.body.string())
+        val pagesArray = json.getJSONArray("pages")
+        val listUrl = (0 until pagesArray.length()).map { i ->
+            pagesArray.getString(i)
+        }
+        return listUrl.mapIndexed { index, url ->
+            Page(index, imageUrl = url)
+        }
```

Comment:
Make a DTO class for this instead of using "org.json".

Reply:
> Make a DTO class for this instead of using "org.json".

That is for sure.